scripts/upload-spotlight.js

```diff
@@ -0,0 +1,267 @@
+/**
+ * Upload Script for SIGNAL's data spotlights
+ *
+ * Command Line Arguments
+ *  Run the script the -h help flag to see up to date descriptions for command line arguments
+ *  node ./scripts/upload-spotlight.js -h
+ *
+ * Example Run
+ * node ./scripts/upload-data.js --id housing-spotlight --doc outcomes --statsfile ./data/vaccine_coldspot_statistics_03_15_2022.json --barriersfile ./data/vaccine_coldspot_side_panel_percentages_03_15_2022.json --statebarriersfile ./data/statebarriers.json --locationsfile ./data/vaccine_coldspot_locations_03_15_2022.json
+ */
+
+/* eslint "@typescript-eslint/no-var-requires": "off" */
+const fs = require("fs");
+const path = require("path");
+const { ArgumentParser } = require("argparse");
+const { initializeApp } = require("firebase-admin/app");
+const { getFirestore } = require("firebase-admin/firestore");
+const { stringify } = require("zipson");
+
+const warnAndExit = (warning) => {
+  console.warn(warning);
+  process.exit(1);
+};
+
+const getJson = (jsonFile) => {
+  const rawdata = fs.readFileSync(jsonFile);
+  return JSON.parse(rawdata);
+};
+
+const argparse = new ArgumentParser({
+  description: "SIGNAL - upload spotlight data to Firestore",
+  add_help: true,
+});
+
+argparse.add_argument("-i", "--id", {
+  required: true,
+  help: "ID of dataset for collection",
+});
+
+argparse.add_argument("-s", "--ageadjusted", {
+  required: false, // TO_REVIEW should this be required?
+  help: "New data for the outcome stats",
+});
+
+argparse.add_argument("-a", "--agespecific", {
+  required: false, // TO_REVIEW should this be required?
+  help: "New data for the affordability outcomes",
+});
+
+argparse.add_argument("-o", "--overwrite", {
+  action: "store_true",
+  help: "if files already exist, overwrite them",
+});
+
+argparse.add_argument("-n", "--newId", {
+  action: "store_true",
+  help: "if the collection id does not exist, creates a new collection",
+});
+
+argparse.add_argument("-z", "--localDir", {
+  help: "path to local folder to save download files to",
+});
+
+const main = async () => {
+  const { id, ageadjusted, agespecific, newId, localDir } =
+    argparse.parse_args();
+
+  const files = [
+    {
+      filePath: ageadjusted,
+      extension: "json",
+      field: "adjusted", // TO_REVIEW what is field? used for localdir, is this the field in the firestore db?
```

Comment:
So the fields here were supposed to be on firebase as to where to put the data. Looks like you won't need this since you're just going to put all the data into the document more directly

Reply:
For example, for the cold spots data, we have fields for barriers etc.

<img width="645" alt="Screen Shot 2022-10-19 at 12 34 56 PM" src="https://user-images.githubusercontent.com/5270855/196751578-03d60a51-5859-4872-9a2f-35d22d1ccd47.png">